### ISSUE TITLE
"ID missing on group record" when retrieving groups

### ISSUE THREAD

**Issue by micw** [AsamK/signal-cli#642]
Hello,

I try to receive all available group. So I request a sync and run receive followed by manager.getGroups(), which usually returned all groups. But this somehow has stopped to work. Instead I get a warning:
```
2021-06-11 21:51:08.890  WARN 13045 --- [           main] org.asamk.signal.manager.Manager         : Failed to handle received sync groups “/tmp/signal-cli_tmp_13973354145050195883.tmp”, ignoring: ID missing on group record!
```

Was something changed in groups by signal?

**Comment by AsamK** [comment]
Seems you received an invalid group in the sync message ... which app/OS are you using for the primary device?
I've added a new check, to at least import the other groups, if one is invalid.

**Comment by micw** [comment]
I'm using Signal Android as primary device and do link to it. I have only one single group for testing. Interestingly, everything worked on my initial attempt. Then I deleted all signal-cli session files, removed the linked devices from my phone and linked again. Not it fails with the error above. I have not changed anything in the groups meanwhile.

May it be that signal had changed? I have seen that there's a newer version of libsignal-service-java available, which uses a _way_ newer version of the signal lib.

**Comment by AsamK** [comment]
Nothing has changed in regard to group sync, the libsignal-client is not involved there.
However the group sync only contains v1 groups, and your test group is most likely a v2 group. So the group sync message is probably empty and maybe libsignal-service-java is not handling empty lists correctly.
To see a v2 group with signal-cli, it needs to receive a group message, then it will be visible in the group list.

**Comment by micw** [comment]
That may be, I indeed received messages from that group on my 1st try.
Is there a good way to reliably receive a list of all groups, even when there's no message in a group? I'd like to 2-way-sync what's app and signal channels but if I see it right, I cannot send to a group before it's known to the client.

**Comment by AsamK** [comment]
There's a new storage system for this, but it's not implemented in signal-cli yet:  #604
